File: Net/Patterns/State/After/NoQuarterState.cs
namespace Patterns.State.After
{
    using System;

    internal class NoQuarterState : IState
    {
        private readonly GumballMachine gumballMachine;

        public NoQuarterState(GumballMachine gumballMachine)
        {
            this.gumballMachine = gumballMachine;
        }

        public void Insert()
        {
            gumballMachine.State = gumballMachine.HasQuarter;
            Console.WriteLine(MachineMessages.InsertSuccessfully);
        }

        public void Eject()
        {
            Console.WriteLine(MachineMessages.EjectWhenNoQuarter);
        }

        public void Turn()
        {
            Console.WriteLine(MachineMessages.TurnWhenNoQuarter);
        }

        public void Dispense()
        {
            Console.WriteLine(MachineMessages.DispenseWhenNoQuarter);
        }
    }
}

File: Java/After/src/main/state/NoQuarterState.java
package state;

public class NoQuarterState implements State {
    GumballMachine gumballMachine;
 
    public NoQuarterState(GumballMachine gumballMachine) {
        this.gumballMachine = gumballMachine;
    }
 
	public void insertQuarter() {
		System.out.println(MachineMessages.InsertSuccessfully);
		gumballMachine.setState(gumballMachine.getHasQuarterState());
	}
 
	public void ejectQuarter() {
		System.out.println(MachineMessages.EjectWhenNoQuarter);
	}
 
	public void turnCrank() {
		System.out.println(MachineMessages.TurnWhenNoQuarter);
	 }
 
	public void dispense() {
		System.out.println(MachineMessages.DispenseWhenNoQuarter);
	} 
}
